make.ps1

```diff
@@ -0,0 +1,280 @@
+    while ($true) {
+        Write-Log ('kubectl get nodes | grep winw1 - attempt {0}' -f $count)
+        if (vagrant ssh controlplane -c 'kubectl get nodes' | Select-String -SimpleMatch 'winw1') {
+            break
+        }
+        Write-Log ('vagrant provision winw1 - attempt {0}' -f $count)
+        vagrant provision winw1
+        $count += 1
+    }
+
+    Write-Log 'Creating .\.lock\joined indicator for Vagrantfile'
+    New-Item -Path '.\.lock\joined' -ItemType File -Force
+    Write-Log ('vagrant provision winw1 - attempt {0}' -f $count)
+    vagrant provision winw1
+
+    Write-Log 'Creating .\.lock\cni indicator for Vagrantfile'
+    New-Item -Path '.\.lock\cni' -ItemType File -Force
+
+    Write-Log 'Cluster created'
+    vagrant status
+    vagrant ssh controlplane -c 'kubectl get nodes'
+}
+
+function script:Invoke-Status {
+    Write-Log 'Invoking command: status'
+
+    Write-Log 'vagrant status'
+    vagrant status
+    
+    Write-Log 'kubectl get nodes'
+    vagrant ssh controlplane -c 'kubectl get nodes'
+}
+#endregion Command Functions
+
+#region Main Script
+$commands = @{
+    'clean'      = '0. Start fresh destroying any existing Vagrant machines';
+    'download'   = '1. Download Kubernetes binaries for Linux and Windows (set version in variables.yaml or variables.local.yaml)';
+    'build'      = '2. Optionally, build Kubernetes from sources for Linux and Windows';
+    'run'        = '3. Create and run two-node cluster';
+    'status'     = '4. Check state of Vagrant machines and Kubernetes nodes';
+    'smoke-test' = '5. Run smoke tests';
+    'e2e-test'   = '6. Run end-to-end tests';
+    'help'       = 'Print this message';
+}
+
+if ($args.Count -eq 0 -or $commands.Keys -notcontains $args[0] -or $args[0] -contains 'help') {
+    Write-Host "Usage: .\make.ps1 <command>"
+    Write-Host "`nRun commands one by one in the following order:"
+    $commands.GetEnumerator() | Sort-Object -Property Value | Format-Table -HideTableHeaders -AutoSize
+    Write-Host "`nDefault settings are defined in variables.yaml file."
+    Write-Host "To tweak the defaults, make a copy as variables.local.yaml and edit.`n"
+    exit
+}
+
+script:Set-VagrantEnv
+
+$command = $args[0]
+if ($command -eq 'clean') {
+    script:Invoke-Clean
+}
+elseif ($command -eq 'status') {
+    script:Invoke-Status
+}
+else {
+    $logTime = (Get-Date -Date $script:startTime -UFormat '%Y%m%d%H%M%S')
+    $logFile = (Join-Path -Path (Get-Location) -ChildPath ('make-{0}-{1}.log' -f $command, $logTime))
+    Write-Log "Saving command output to $logFile"
+    
+    if ($command -eq 'download') {
+        script:Invoke-Download | Tee-Object -FilePath $logFile
+    }
+    elseif ($command -eq 'build') {
+        Write-Host 'TODO: Invoke-Build'
+        #script:Invoke-Build | Tee-Object -FilePath $logFile
+    }
+    elseif ($command -eq 'run') {
+        script:Invoke-Run | Tee-Object -FilePath $logFile
+    }
+    elseif ($command -eq 'smoke-test') {
```

Comment:
can you open an issue for each of these TODO implementations?

Reply:
I have just committed invocations of commands running tests

Reply:
Here is the `TODO: Invoke-Build` item copied
- #255